no pthread_spinlock_t on mac

New Issue for sst-macro
-----------------------

1 - Detailed description of problem or enhancement
Builds using --enable-spinlock fail on OSX since some functionality lives under different names: 
```
In file included from ../../../sst-macro/sstmac/common/runtime.cc:53:
In file included from ../../../sst-macro/sstmac/backends/common/parallel_runtime.h:51:
../../../sst-macro/sstmac/common/thread_lock.h:101:3: error: unknown type name 'pthread_spinlock_t'; did you mean 'pthread_rwlock_t'?
  pthread_spinlock_t lock_;
  ^~~~~~~~~~~~~~~~~~
  pthread_rwlock_t
/usr/include/sys/_pthread/_pthread_rwlock_t.h:30:35: note: 'pthread_rwlock_t' declared here
typedef __darwin_pthread_rwlock_t pthread_rwlock_t;
                                  ^
1 error generated.
```

2 - Describe how to reproduce
Try to configure with --enable-spinlock on OSX 

3 - What Operating system(s) and versions 
OS X

4 - What version of external libraries (Boost, MPI)
Neither 

5 - Provide sha1 of all relevant sst repositories (sst-core, sst-elements, etc)
sst-macro -> b14225b51c09c7e6ea3ad04ddbc98f7a2deda342

This might already be a known issue, but I didn't find it by searching.

--- a/acinclude/check_thread.m4
+++ b/acinclude/check_thread.m4
@@ -18,7 +18,11 @@ AC_ARG_ENABLE([spinlock],
   [with_spinlock=$enableval],
   [with_spinlock=no])
 if test "X$with_spinlock" = "Xyes"; then
-  AC_DEFINE_UNQUOTED([USE_SPINLOCK], 1, "Whether to use spin locks for more efficient thread barriers")
+  if test "X$darwin" = "Xfalse"; then
+    AC_DEFINE_UNQUOTED([USE_SPINLOCK], 1, "Whether to use spin locks for more efficient thread barriers")
+  else
+    AC_MSG_ERROR([--enable-spinlock not available on darwin])
+  fi
 fi
 
 # pthread_setaffinity_np checking: from https://github.com/landonf/haiku-buildtools/blob/master/gcc/libgomp/configure.ac

